<?xml version="1.0"?>
<launch>

    <arg name="veh" default="$(env VEHICLE_NAME)"/>
    <arg name="camera_name" default="jetson_camera"/>
    <arg name="pkg_name" value="jetson_camera"/>
    <arg name="config" default="baseline"/>
    <arg name="cam_info" default="true" doc="If set to true, use cam_info_reader_node to pusblish CameraInfo."/>
    <arg name="raw" default="false" doc="If set to true, also publish raw image using decoder." />
    <arg name="rect" default="false" />
    <arg name="live" default="true" doc="live set to false means we don't run the actual camera (i.e. running from a log file" />
    <arg name="pub_freq" default="1"/>
    
    <!-- jetson_camera -->
    <group if="$(arg live)">
        <!-- original topic : $(arg camera_name)/image_raw/compressed -->
        <node ns="$(arg veh)" pkg="jetson_camera" type="jetson_camera.py" name="$(arg camera_name)" clear_params="true" output="screen">
            <rosparam file="$(find jetson_camera)/param/jetson_camera.yaml" command="load"/>
        </node>
    </group>

    
    <!-- decoder_node -->
    <group if="$(arg raw)">
        <remap from="/$(arg veh)/decoder_node/compressed_image" to="/$(arg veh)/$(arg camera_name)/image/compressed" />
        <remap from="/$(arg veh)/decoder_node/image/raw" to="/$(arg veh)/$(arg camera_name)/image/raw" />
        <include file="$(find jetson_camera)/launch/decoder_node.launch">
            <arg name="veh" value="$(arg veh)"/>
            <arg name="config" value="$(arg config)"/>
            <arg name="param_file_name" value="$(arg veh)"/>
            <arg name="pkg_name" value="$(arg camera_name)"/>
        </include>
    </group>

    <!-- cam_info_reader_node -->
    <group if="$(arg cam_info)"> 
        <remap from="cam_info_reader_node/camera_info" to="$(arg camera_name)/camera_info"/>
        <remap from="cam_info_reader_node/compressed_image" to="/$(arg veh)/$(arg camera_name)/image/compressed"/>
        <include file="$(find jetson_camera)/launch/cam_info_reader_node.launch"> 
            <arg name="veh" value="$(arg veh)"/>
            <arg name="config" value="$(arg config)"/>
            <arg name="pub_freq" value="60"/>
        </include>
        
         <group if="$(arg raw)">
             <remap from="raw_cam_info_reader_node/camera_info" to="$(arg camera_name)/raw_camera_info"/>
             <remap from="raw_cam_info_reader_node/compressed_image" to="$(arg camera_name)/image/raw"/>
             <include file="$(find jetson_camera)/launch/cam_info_reader_node.launch">
                 <arg name="veh" value="$(arg veh)"/>
                 <arg name="config" value="$(arg config)"/>
                 <arg name="node_name" value="raw_cam_info_reader_node" />
                 <arg name="image_type" value="raw" /> 
             </include>                
         </group>                 <!-- group for raw -->
     </group>                     <!-- group for cam_info -->
</launch>



    
 
